### DISCUSSION TITLE
exploring-dotnet-6-part-9-source-generator-updates-incremental-generators/

### DISCUSSION THREAD

**Discussion by andrewlock** [andrewlock/blog-comments#365] [Announcements]
# exploring-dotnet-6-part-9-source-generator-updates-incremental-generators/

In this post I look at the updates to the source generator API in .NET 6, why the changes were made, and how to update your source generators to use them.

https://andrewlock.net/exploring-dotnet-6-part-9-source-generator-updates-incremental-generators/

[a comment by an automated account is not shown]

**Comment by cammerman** [comment]
The final code sample suddenly references a Parser class and an Emitter class which are not mentioned anywhere else, and do not appear to be in any of the namespaces that are used for anything else. What are these classes? Where do they come from? Are these specific to a particular version of .NET or Roslyn? Are they an exercise left to the reader? What are these?
